tests/draft-next/dynamicRef.json

```diff
@@ -612,5 +612,31 @@
                 "valid": false
             }
         ]
+    },
+    {
+        "description": "$dynamicRef points to a boolean schema",
+        "schema": {
+            "$defs": {
+                "boolean": true
+            },
+            "properties": {
+                "true": {
+                    "$dynamicRef": "#/$defs/boolean"
+                },
+                "false": false
+            }
+        },
+        "tests": [
+            {
+                "description": "follow $dynamicRef to a true schema",
+                "data": { "true": 1 },
+                "valid": true
+            },
+            {
+                "description": "false schema",
+                "data": { "false": 1 },
```

Comment:
This branch of the test seems "uninteresting" / unrelated to this test case -- I assume it fails both before and after whatever bugfix you made. It seems more likely to be interesting/relevant if you put it behind a dynamic ref itself? i.e. `#/$defs/true` and `#/$defs/false` and fail when you hit `false` via the dynamicRef.

Reply:
The intent is to show a way to a 'valid' state. Otherwise, a naive interpretation of the test should be that this input should always cause a validation failure. Do you have a suggestion for a better way to provide a valid data input?

Reply:
> The intent is to show a way to a 'valid' state. 

(I was commenting on the *invalid* branch, which fails simply because it's equivalent to `{"properties": {"foo": false}}`)

What I was trying to suggest instead above (albeit I don't know whether it exercises your specific bug you were targeting, but it seems like it should be the same) was to use this schema:

```json
{
            "$defs": {
                "true": true,
                "false": false
            },
            "properties": {
                "true": {
                    "$dynamicRef": "#/$defs/true"
                },
                "false": { "$dynamicRef": "#/$defs/false" }
            }
        }
```

and to then have the two instances (and outcomes) be exactly the ones you have now I think.